src/main/scala/is/hail/variant/GenomeReference.scala

```diff
@@ -157,16 +157,30 @@ case class GenomeReference(name: String, contigs: Array[String], lengths: Map[St
     }
   }
 
-  def unify(concrete: GenomeReference): Boolean = this == concrete
+  def unify(concrete: GRBase): Boolean = this == concrete
```

Comment:
There can only be one reference object for each reference, right?  You might change this to an object reference equality check: `this eq concrete`.

Reply:
Does calling `GenomeReference.GRCh37` generate a new GenomeReference class instance each time?